src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,670 @@
+}
+
+if (window.onanimationend === undefined && window.onwebkitanimationend !== undefined) {
+  CSS_PREFIX = '-webkit-';
+  ANIMATION_PROP = 'WebkitAnimation';
+  ANIMATIONEND_EVENT = 'webkitAnimationEnd animationend';
+} else {
+  ANIMATION_PROP = 'animation';
+  ANIMATIONEND_EVENT = 'animationend';
+}
+
+var DURATION_KEY = 'Duration';
+var PROPERTY_KEY = 'Property';
+var DELAY_KEY = 'Delay';
+var TIMING_KEY = 'TimingFunction';
+var ANIMATION_ITERATION_COUNT_KEY = 'IterationCount';
+var ANIMATION_PLAYSTATE_KEY = 'PlayState';
+var ELAPSED_TIME_MAX_DECIMAL_PLACES = 3;
+var CLOSING_TIME_BUFFER = 1.5;
+var ONE_SECOND = 1000;
+var BASE_TEN = 10;
+
+function computeCSSStyles($window, element, properties) {
+  var styles = {};
+
+  var detectedStyles = $window.getComputedStyle(element) || {};
+  forEach(properties, function(style, prop) {
+    var val = detectedStyles[style];
+    // only numerical-based values have a digit as the first value
+    if (val) {
+      if (val.charAt(0) >= 0) {
+        val = parseMaxTime(val);
+      }
+      styles[prop] = val;
+    }
+  });
+
+  return styles;
+}
+
+function parseMaxTime(str) {
+  var maxValue = 0;
+  var values = isString(str) ?
+    str.split(/\s*,\s*/) :
+    [];
+  forEach(values, function(value) {
+    maxValue = Math.max(parseFloat(value) || 0, maxValue);
+  });
+  return maxValue;
+}
+
+function getCssTransitionDurationStyle(duration, applyOnlyDuration) {
+  var style = TRANSITION_PROP;
+  var value = duration + 's';
+  if (applyOnlyDuration) {
+    style += DURATION_KEY;
+  } else {
+    value += ' linear all';
+  }
+  return [style, value];
+}
+
+function getCssKeyframeDurationStyle(duration) {
+  return [ANIMATION_PROP + DURATION_KEY, duration + 's'];
+}
+
+function getCssDelayStyle(delay, isKeyframeAnimation) {
+  var prop = (isKeyframeAnimation ? ANIMATION_PROP : TRANSITION_PROP) + DELAY_KEY;
+  return [prop, delay + 's'];
+}
+
+function blockTransitions(node, bool) {
+  var value = bool ? 'none' : '';
+  var key = TRANSITION_PROP + PROPERTY_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function blockAnimations(node, bool) {
+  var value = bool ? 'paused' : '';
+  var key = ANIMATION_PROP + ANIMATION_PLAYSTATE_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function applyInlineStyle(node, key, value) {
+  if (arguments.length == 2) {
+    value = key[1];
+    key = key[0];
+  }
+  node.style[key] = value;
+}
+
+function LocalCacheLookup() {
+  var cache = {};
+  return {
+    flush: function() {
+      cache = {};
+    },
+
+    count: function(key) {
+      var entry = cache[key];
+      return entry ? entry.total : 0;
+    },
+
+    get: function(key) {
+      var entry = cache[key];
+      return entry && entry.value;
+    },
+
+    put: function(key, value) {
+      if (!cache[key]) {
+        cache[key] = { total: 0 };
+      }
+      cache[key].total++;
+      cache[key].value = value;
+    }
```

Comment:
This could be optimized, since we appear to never change the value, just increment it.

```js
   put: function(key, value) {
     if (!cache[key]) {
       cache[key] = { total: 1, value: value };
     } else {
       cache[key].total++;
    }
   }
```